### PR TITLE
[ADD] purchase_exception to 11.0

### PR DESCRIPTION
**Pull request by mourad-ehm**
This module add exception management to purchase (based on base_exception).
A user can add new exceptions to block purchase if qty is < 0 for example.
 Depends on:
- [x] base_exception https://github.com/OCA/server-tools/pull/1025

### CODE REVIEW

File: purchase_exception/data/purchase_exception_data.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            <field name="interval_number">20</field>
+            <field name="interval_type">minutes</field>
+            <field name="numbercall">-1</field>
+            <field name="doall" eval="False" />
+            <field name="active" eval="False" />
+        </record>
+
+        <record id ="po_excep_no_email" model="exception.rule">
+            <field name="name">No email on vendor</field>
+            <field name="description">No Email for Vondor</field>
```
Comment (rousseldenis) [inline purchase_exception/data/purchase_exception_data.xml]: Vendor

**Review by rousseldenis — COMMENTED** [review]
LGTM. Typo

File: oca_dependencies.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+server-tools https://github.com/akretion/server-tools 11.0-mig-base_exception
```
Comment (lasley) [inline oca_dependencies.txt]: What pull request does this depend on?

Comment (JordiBForgeFlow) [reply oca_dependencies.txt]: OCA/server-tools#1025

**Comment by bealdav** [comment]
result log in runbot is

> 2017-11-23 09:14:27,108 159 INFO openerp_test odoo.modules.loading: Modules loaded.
2017-11-23 09:14:27,111 159 INFO openerp_test odoo.models: Computing parent left and right for table ir_ui_menu...
2017-11-23 09:14:27,155 159 INFO openerp_test odoo.service.server: Initiating shutdown
2017-11-23 09:14:27,155 159 INFO openerp_test odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.
----------
> Module test summary
>[1;32mSuccess[0;m purchase_exception

+=======================================
|  Tests summary:
|---------------------------------------
| test_server.py              [1;32mSuccess[0;m
+=======================================


Then I think it's a false fail.

What do you think @jbeficent ?

**Review by bealdav — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@bealdav @mourad-ehm See https://github.com/akretion/purchase-workflow/pull/5

**Comment by bealdav** [comment]
Yet runbot false negative

> 2017-11-23 15:14:49,124 159 INFO openerp_test odoo.modules.loading: 35 modules loaded in 1.31s, 9 queries
> 2017-11-23 15:14:49,659 159 INFO openerp_test odoo.modules.loading: Modules loaded.
> 2017-11-23 15:14:49,663 159 INFO openerp_test odoo.models: Computing parent left and right for table ir_ui_menu...
> 2017-11-23 15:14:49,706 159 INFO openerp_test odoo.service.server: Initiating shutdown
> 2017-11-23 15:14:49,706 159 INFO openerp_test odoo.service.server: Hit CTRL-C again or send a second signal to force the shutdown.

----------
Module test summary
[1;32mSuccess[0;m purchase_exception

+=======================================
|  Tests summary:
|---------------------------------------
| test_server.py              [1;32mSuccess[0;m
+=======================================

**Comment by rousseldenis** [comment]
@bealdav Nope. Warning:

Field base.exception.test.model.line.lead_id with unknown comodel_name 'base.exception.test.model'

**Comment by rousseldenis** [comment]
@bealdav Due to base_exception module (not merged yet)

**Comment by lasley** [comment]
@mourad-ehm - base_exception has been merged. Please rebase 🚀

**Comment by JordiBForgeFlow** [comment]
@mourad-ehm - can you rebase? I'm depending on this PR to merge other modules.

**Comment by JordiBForgeFlow** [comment]
@mourad-ehm PR OCA/server-tools#1111 fixes the problem in base_exception

**Comment by JordiBForgeFlow** [comment]
You can see how the test is green in PR https://github.com/OCA/purchase-workflow/pull/487

**Comment by mourad-ehm** [comment]
Thanks @jbeficent for the correction. Sorry for my late response. I'm on vacation.

**Comment by JordiBForgeFlow** [comment]
merged in https://github.com/OCA/purchase-workflow/pull/487

**Pull request closed without merging**